### ISSUE TITLE
Doesn't work locally

### ISSUE THREAD

**Issue by axljones** [watsonbox/exportify#146]
When running the project locally, it doesn't work as my internal ip is passed to the spotify API:

ERROR:
INVALID_CLIENT: Invalid redirect URI

URI:
https://accounts.spotify.com/authorize?client_id=abcd&redirect_uri=http%3A%2F%2F192.168.0.99%3A3000%2Fexportify&scope=playlist-read-private%20playlist-read-collaborative%20user-library-read&response_type=token&show_dialog=false

**Comment by fcivaner** [comment]
Recently tested using docker on my local, I weren't able to reproduce this issue with Firefox 115.3.0esr (64-bit). Just writing to provide additional information,
PR for docker usage: https://github.com/watsonbox/exportify/pull/148

**Comment by axljones** [comment]
> Recently tested using docker on my local, I wasn't able to reproduce this issue with Firefox 115.3.0esr (64-bit). Just writing to provide additional information, PR for docker usage: #148

Did you go through with the import? 
Did you run this on a server with a publicly reachable ip address?

**Comment by fcivaner** [comment]
I ran it on my local PC, using;

`docker build . -t exportify

docker run -p 3000:3000 exportify`

And then went to http://localhost:3000/ on my browser. Then I exported a couple of playlists, the data was correct

**Comment by axljones** [comment]
> I ran it on my local PC, using;
> 
> `docker build . -t exportify
> 
> docker run -p 3000:3000 exportify`
> 
> And then went to http://localhost:3000/ on my browser. Then I exported a couple of playlists, the data was correct

The dockerfile didn't exist when I tried last time.

 I've tried it again and get the same issue. Could you give me the redirect URI you get ?
 
 Especially the value of the redirect_uri parameter.

**Comment by fcivaner** [comment]
It's in the form of

```https://accounts.spotify.com/en/authorize?client_id=[clientid]&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2F&scope=playlist-read-private%20playlist-read-collaborative%20user-library-read&response_type=token&show_dialog=false```

Maybe spotify makes an exception for localhost when handling these requests, but not for the IP address of a local server, if you are using an IP address in the local network?

**Comment by axljones** [comment]
> It's in the form of
> 
> `https://accounts.spotify.com/en/authorize?client_id=[clientid]&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2F&scope=playlist-read-private%20playlist-read-collaborative%20user-library-read&response_type=token&show_dialog=false`
> 
> Maybe spotify makes an exception for localhost when handling these requests, but not for the IP address of a local server, if you are using an IP address in the local network?

Yes, that is exactly what i was thinking originally. I tried using localhost but it didn't work because i didn't specify the port. It worked with your redirect_uri. I then got a new redirect url with a token so i replaced the localhost with my ip. It now loads but seems stuck in a loop. I'll keep this issue updated


http://localhost:3000/exportify#access_token=[access_token]&token_type=Bearer&expires_in=3600

![image](https://github.com/watsonbox/exportify/assets/24891578/11803f1f-197f-4f04-aaf6-9516ebf569dc)

**Comment by awsms** [comment]
both selfhost methods (docker & running it directly with yarn) did not work for me. the service hosted on github.io doesn't work either though :(

**Comment by watsonbox** [comment]
Converting this to a discussion in case it evolved. I can only guess that there must be something unusual about your browser or extensions causing this issue, since that local redirect URI is valid for the app and works for others.
